src/sentry/api/endpoints/organization_event_details.py

```diff
@@ -86,16 +87,43 @@ class OrganizationEventDetailsEndpoint(OrganizationEventsEndpointBase):
         "GET": ApiPublishStatus.PRIVATE,
     }
 
-    def get(self, request: Request, organization, project_slug, event_id) -> Response:
-        """event_id is validated by a regex in the URL"""
-        if not self.has_feature(organization, request):
-            return Response(status=404)
+    def convert_args(
+        self,
+        request: Request,
+        organization_slug: str | int | None = None,
+        *args: Any,
+        **kwargs: Any,
+    ) -> tuple[tuple[Any, ...], dict[str, Any]]:
+        args, kwargs = super().convert_args(request, organization_slug, *args, **kwargs)
+
+        organization = kwargs["organization"]
+        project_slug = kwargs.pop("project_slug")
 
         try:
-            project = Project.objects.get(
-                slug=project_slug, organization_id=organization.id, status=ObjectStatus.ACTIVE
-            )
+            if id_or_slug_path_params_enabled(
+                convert_args_class=self.convert_args.__qualname__,
+                organization_slug=organization.slug,
+            ):
+                project = Project.objects.get(
+                    slug__id_or_slug=project_slug,
+                    organization_id=organization.id,
+                    status=ObjectStatus.ACTIVE,
+                )
+            else:
+                project = Project.objects.get(
+                    slug=project_slug, organization_id=organization.id, status=ObjectStatus.ACTIVE
+                )
+
+            kwargs["project"] = project
+
         except Project.DoesNotExist:
+            raise Project.DoesNotExist
```

Comment:
is this necessary? are you just raising the caught exception?

Reply:
whoops, I was working too fast

Reply:
fixed this up